flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/fs/ForStFlinkFileSystem.java

```diff
@@ -305,7 +306,8 @@ public synchronized FileStatus getFileStatus(Path path) throws IOException {
         if (mappingEntry == null) {
             return new FileStatusWrapper(delegateFS.getFileStatus(path), path);
         }
-        if (FileOwnershipDecider.shouldAlwaysBeLocal(path)) {
+        if (FileOwnershipDecider.shouldAlwaysBeLocal(path)
+                && mappingEntry.getFileOwnership() != FileOwnership.NOT_OWNED) {
```

Comment:
I see in the Jira it says:
"When restored, the file is existed, but fails because it always try to find a sst in the local filesystem:"
I do not follow this. Why is `FileOwnershipDecider.shouldAlwaysBeLocal(path)` true for a non local file?
When this method is explicitly checking whether the file is an sst file?
```
public static boolean shouldAlwaysBeLocal(Path filePath) {
 return !isSstFile(filePath);
}
```

Would it be possible to ensure earlier in the code that an sst file is always seen as an sst file? So the original  check works as expected.

Reply:
I just put the judgement into the new method of `shouldAlwaysBeLocal`.
The NOT_ONWED sst file will always on the remote, so it should not be local.
The file ownership always exists expect new created file or restored file which will be determined by the file format like '.sst'

Reply:
Hi @davidradl , Do you have other comments?